jkube-kit/jkube-kit-wildfly-jar/src/test/java/org/eclipse/jkube/wildfly/jar/generator/WildflyJARGeneratorTest.java

```diff
@@ -53,7 +64,204 @@ public void getEnv() throws IOException {
+        Map<String, String> pluginOptions = new HashMap();
+        Path tmpDir = Files.createTempDirectory("bootable-jar-test-project2");
+        Path targetDir = tmpDir.resolve("target");
+        Path repoDir = targetDir.resolve("myrepo");
+        Files.createDirectories(repoDir);
+        options.put(PLUGIN_OPTIONS, pluginOptions);
+        pluginOptions.put(JBOSS_MAVEN_DIST, null);
+        pluginOptions.put(JBOSS_MAVEN_REPO, repoDir.toString());
+        try {
+            GeneratorContext ctx = contextForSlimServer(project, options, null);
+            WildflyJARGenerator generator = new WildflyJARGenerator(ctx);
+            List<String> extraOptions = generator.getExtraJavaOptions();
+            assertNotNull(extraOptions);
+            assertEquals(2, extraOptions.size());
+            assertEquals("-Djava.net.preferIPv4Stack=true", extraOptions.get(0));
+            assertEquals("-Dmaven.repo.local=/deployments/myrepo", extraOptions.get(1));
+            List<AssemblyFileSet> files = generator.addAdditionalFiles();
+            assertFalse(files.isEmpty());
+            AssemblyFileSet set = files.get(files.size() - 1);
+            assertEquals(targetDir.toFile(), set.getDirectory());
+            assertEquals(1, set.getIncludes().size());
+            assertEquals("myrepo", set.getIncludes().get(0));
+        } finally {
+            Files.delete(repoDir);
+            Files.delete(targetDir);
+            Files.delete(tmpDir);
+        }
+    }
+    
+    @Test
+    public void slimServerNoDir(@Mocked final JavaProject project) throws Exception {
+        Map<String, Object> options = new HashMap<>();
+        Map<String, String> pluginOptions = new HashMap();
+        Path tmpDir = Files.createTempDirectory("bootable-jar-test-project2");
+        Path targetDir = tmpDir.resolve("target");
+        Path repoDir = targetDir.resolve("myrepo");
+        options.put(PLUGIN_OPTIONS, pluginOptions);
+        pluginOptions.put(JBOSS_MAVEN_DIST, null);
+        pluginOptions.put(JBOSS_MAVEN_REPO, repoDir.toString());
+        try {
+            GeneratorContext ctx = contextForSlimServer(project, options, null);
+            WildflyJARGenerator generator = new WildflyJARGenerator(ctx);
+            List<String> extraOptions = generator.getExtraJavaOptions();
+            assertNotNull(extraOptions);
+            assertEquals(2, extraOptions.size());
+            assertEquals("-Djava.net.preferIPv4Stack=true", extraOptions.get(0));
+            assertEquals("-Dmaven.repo.local=/deployments/myrepo", extraOptions.get(1));
+            boolean failed = true;
+            try {
+                List<AssemblyFileSet> files = generator.addAdditionalFiles();
+                failed = false;
+            } catch(Exception ex) {
+                // XXX Expected
+            }
+            if (!failed) {
+                throw new Exception("Test should have failed, no directory for maven repo");
+            }
```

Comment:
```suggestion
            /* Exception result = */assertThrows(Exception.class, () -> {
              generator.addAdditionalFiles();
              fail("Test should have failed, no directory for maven repo");
            });
            // Optionally assert exception e.g. assertEquals("Expected Message", result.getMessage());
```

Reply:
@manusa , thank-you, I integrated your suggestion.